src/user.controller.ts

```diff
@@ -37,10 +42,26 @@ export class UserController {
         }
 
         try {
-            const savedUser = await this.repository.save(newUser);
+            const hashuser = {
+                key: (bcrypt.hashSync(Date.toString() + newUser.name, this.PASSWORD_HASH_SIZE)).replace(".", "").replace("\/", "").replace("&", ""),
+                user: newUser,
+                creationDate: new Date(),
+            }
+            var userId = 0;
+            await getManager().transaction(async transactionalEntityManager => {
+                const savedUser: any = await transactionalEntityManager.getRepository(User).save(newUser);
+                hashuser.user = savedUser;
+                userId = savedUser.id;
+                await transactionalEntityManager.getRepository(HashUser).save(hashuser);
+            });
+            try {
+                new RabbitMq("user_created", JSON.stringify({ "destination": [newUser.email], "user_name": newUser.name, "key": hashuser.key }));
+            } catch (error) {
+                console.log(error);
```

Comment:
Here we should return a response and stop the method execution

Reply:
Done